File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ListBox/Functionality/Transfer.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;
using System.Collections;

#endregion

namespace CompanionKit.Controls.ListBox.Functionality
{
    public partial class Transfer : UserControl
    {
        public Transfer()
        {
            InitializeComponent();
        }

        private void btnToRight_Click(object sender, EventArgs e)
        {
            //Add selected items of the left ListBox into the right ListBox
            this.mobjListBox2.Items.AddRange(this.mobjListBox1.SelectedItems);

            //Remove selected items from the left ListBox 
            string[] selectedItems = new string[this.mobjListBox1.SelectedItems.Count];
            this.mobjListBox1.SelectedItems.CopyTo(selectedItems, 0);
            foreach (object selectedItem in selectedItems)
            {
                this.mobjListBox1.Items.Remove(selectedItem);
            }
            
            UpdatedEnableStatusForTransferedButtons();
            
        }

        /// <summary>
        /// Handles the Click event of the btnAllToRight control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void btnAllToRight_Click(object sender, EventArgs e)
        {
            //Add all items of the left ListBox into the right ListBox
            this.mobjListBox2.Items.AddRange(this.mobjListBox1.Items);
            //Remove all items from the left ListBox 
            this.mobjListBox1.Items.Clear();
            UpdatedEnableStatusForTransferedButtons();
        }

        /// <summary>
        /// Handles the Click event of the btnToLeft control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void btnToLeft_Click(object sender, EventArgs e)
        {
            //Add selected items of the right ListBox into the left ListBox
            this.mobjListBox1.Items.AddRange(this.mobjListBox2.SelectedItems);
            
            //Remove selected items from the right ListBox 
            string[] selectedItems = new string[this.mobjListBox2.SelectedItems.Count];
            this.mobjListBox2.SelectedItems.CopyTo(selectedItems, 0);
            foreach (object selectedItem in selectedItems)
            {
                this.mobjListBox2.Items.Remove(selectedItem);
            }
            UpdatedEnableStatusForTransferedButtons();
        }

        /// <summary>
        /// Handles the Click event of the btnAllToLeft control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void btnAllToLeft_Click(object sender, EventArgs e)
        {
            //Add all items of the right ListBox into the left ListBox
            this.mobjListBox1.Items.AddRange(this.mobjListBox2.Items);
            //Remove all items from the left ListBox 
            this.mobjListBox2.Items.Clear();
            UpdatedEnableStatusForTransferedButtons();
        }

        /// <summary>
        /// Handles the Load event of the Transfer control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void Transfer_Load(object sender, EventArgs e)
        {
            //Set initial enable state for buttons that transfer 
            //item between left and right ListBoxes
            UpdatedEnableStatusForTransferedButtons();
        }

        /// <summary>
        /// Updates enable state for buttons that transfer item between left and right ListBoxes
        /// </summary>
        private void UpdatedEnableStatusForTransferedButtons()
        {
            if (mobjListBox1.Items.Count > 0 && !this.btnAllToRight.Enabled)
            {
                this.btnAllToRight.Enabled = true;
            }

            if (mobjListBox1.Items.Count < 1 && this.btnAllToRight.Enabled)
            {
                this.btnAllToRight.Enabled = false;
            }

            if (mobjListBox1.SelectedItems.Count > 0 && !this.btnToRight.Enabled)
            {
                this.btnToRight.Enabled = true;
            }

            if (mobjListBox1.SelectedItems.Count < 1 && this.btnToRight.Enabled)
            {
                this.btnToRight.Enabled = false;
            }

            if (mobjListBox2.Items.Count > 0 && !this.btnAllToLeft.Enabled)
            {
                this.btnAllToLeft.Enabled = true;
            }

            if (mobjListBox2.Items.Count < 1 && this.btnAllToLeft.Enabled)
            {
                this.btnAllToLeft.Enabled = false;
            }

            if (mobjListBox2.SelectedItems.Count > 0 && !this.btnToLeft.Enabled)
            {
                this.btnToLeft.Enabled = true;
            }

            if (mobjListBox2.SelectedItems.Count < 1 && this.btnToLeft.Enabled)
            {
                this.btnToLeft.Enabled = false;
            }
        }

        /// <summary>
        /// Handles the SelectedIndexChanged event of the listBox1 control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjListBox1_SelectedIndexChanged(object sender, EventArgs e)
        {
            //Set initial enable state for buttons that transfer 
            //item between left and right ListBoxes
            UpdatedEnableStatusForTransferedButtons();

        }

        /// <summary>
        /// Handles the SelectedIndexChanged event of the listBox2 control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjListBox2_SelectedIndexChanged(object sender, EventArgs e)
        {
            //Set initial enable state for buttons that transfer 
            //item between left and right ListBoxes
            UpdatedEnableStatusForTransferedButtons();
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ListBox/Functionality/Transfer.vb
Namespace CompanionKit.Controls.ListBox.Functionality

	Public Class Transfer

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()



		End Sub

        ''' <summary>
        ''' Handles the Click event of the BtnAllToLeft control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        Private Sub btnAllToLeft_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnAllToLeft.Click
            'Add all items of the right ListBox into the left ListBox
            Me.mobjListBox1.Items.AddRange(Me.mobjListBox2.Items)
            'Remove all items from the left ListBox 
            Me.mobjListBox2.Items.Clear()
            Me.UpdatedEnableStatusForTransferedButtons()

        End Sub

        ''' <summary>
        ''' Handles the Click event of the BtnToLeft control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        Private Sub btnToLeft_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnToLeft.Click
            'Add selected items of the right ListBox into the left ListBox
            Me.mobjListBox1.Items.AddRange(Me.mobjListBox2.SelectedItems)
            'Remove selected items from the right ListBox 
            Dim selectedItems As String() = New String(Me.mobjListBox2.SelectedItems.Count - 1) {}
            Me.mobjListBox2.SelectedItems.CopyTo(selectedItems, 0)
            Dim selectedItem As String
            For Each selectedItem In selectedItems
                Me.mobjListBox2.Items.Remove(selectedItem)
            Next
            Me.UpdatedEnableStatusForTransferedButtons()

        End Sub

        ''' <summary>
        ''' Handles the Click event of the BtnAllToRight control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        Private Sub btnAllToRight_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnAllToRight.Click
            'Add selected items of the left ListBox into the right ListBoxAdd all items of the left ListBox into the right ListBox
            Me.mobjListBox2.Items.AddRange(Me.mobjListBox1.Items)
            'Remove all items from the left ListBox 
            Me.mobjListBox1.Items.Clear()
            Me.UpdatedEnableStatusForTransferedButtons()

        End Sub

        ''' <summary>
        ''' Handles the Click event of the BtnToRight control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        Private Sub btnToRight_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnToRight.Click
            'Add selected items of the left ListBox into the right ListBox
            Me.mobjListBox2.Items.AddRange(Me.mobjListBox1.SelectedItems)
            'Remove selected items from the left ListBox 
            Dim selectedItems As String() = New String(Me.mobjListBox1.SelectedItems.Count - 1) {}
            Me.mobjListBox1.SelectedItems.CopyTo(selectedItems, 0)
            Dim selectedItem As String
            For Each selectedItem In selectedItems
                Me.mobjListBox1.Items.Remove(selectedItem)
            Next
            Me.UpdatedEnableStatusForTransferedButtons()

        End Sub

        ''' <summary>
        ''' Handles the SelectedIndexChanged event of the mobjListBox1 control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        Private Sub mobjListBox1_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mobjListBox1.SelectedIndexChanged
            'Set initial enable state for buttons that transfer 
            'item between left and right ListBoxes
            Me.UpdatedEnableStatusForTransferedButtons()
        End Sub

        ''' <summary>
        ''' Handles the SelectedIndexChanged event of the mobjListBox2 control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        Private Sub mobjListBox2_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mobjListBox2.SelectedIndexChanged
            'Set initial enable state for buttons that transfer 
            'item between left and right ListBoxes
            Me.UpdatedEnableStatusForTransferedButtons()
        End Sub

        ''' <summary>
        ''' Handles the Load event of the Transfer control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
		Private Sub Transfer_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
			'Set initial enable state for buttons that transfer 
			'item between left and right ListBoxes
			Me.UpdatedEnableStatusForTransferedButtons()
		End Sub

		''' <summary>
		''' Updates enable state for buttons that transfer item between left and right ListBoxes
		''' </summary>
		''' <remarks></remarks>
		Private Sub UpdatedEnableStatusForTransferedButtons()
            If Not ((Me.mobjListBox1.Items.Count <= 0) OrElse Me.btnAllToRight.Enabled) Then
                Me.btnAllToRight.Enabled = True
            End If
            If ((Me.mobjListBox1.Items.Count < 1) AndAlso Me.btnAllToRight.Enabled) Then
                Me.btnAllToRight.Enabled = False
            End If
            If Not ((Me.mobjListBox1.SelectedItems.Count <= 0) OrElse Me.btnToRight.Enabled) Then
                Me.btnToRight.Enabled = True
            End If
            If ((Me.mobjListBox1.SelectedItems.Count < 1) AndAlso Me.btnToRight.Enabled) Then
                Me.btnToRight.Enabled = False
            End If
            If Not ((Me.mobjListBox2.Items.Count <= 0) OrElse Me.btnAllToLeft.Enabled) Then
                Me.btnAllToLeft.Enabled = True
            End If
            If ((Me.mobjListBox2.Items.Count < 1) AndAlso Me.btnAllToLeft.Enabled) Then
                Me.btnAllToLeft.Enabled = False
            End If
            If Not ((Me.mobjListBox2.SelectedItems.Count <= 0) OrElse Me.btnToLeft.Enabled) Then
                Me.btnToLeft.Enabled = True
            End If
            If ((Me.mobjListBox2.SelectedItems.Count < 1) AndAlso Me.btnToLeft.Enabled) Then
                Me.btnToLeft.Enabled = False
            End If
		End Sub

	End Class

End Namespace